inst/tests/tests.Rraw

```diff
@@ -20596,3 +20597,29 @@ test(2295.3, is.data.table(d2))
 
 # #6588: .checkTypos used to give arbitrary strings to stopf as the first argument
 test(2296, d2[x %no such operator% 1], error = '%no such operator%')
+
+# coerce Dates to double if join on multiple columns, #6602
+x = data.table(a=1L)
+y = data.table(c=1L, d=1)
+test(2297.01, options=c(datatable.verbose=TRUE), y[x, on=.(c == a, d == a)], data.table(c=1L, d=1L), output="Coercing .*c to type double")
```

Comment:
nit: WDYT about using `verbose=TRUE` argument to `[` instead?

Reply:
`a` also gets coerced, right? should that be reflected in the `output=` test?

Reply:
(also, I might expect `d` to be the coerced column here...)

Reply:
done